yarn-project/end-to-end/src/e2e_static_calls.test.ts

```diff
@@ -1,34 +1,63 @@
 import { type Wallet } from '@aztec/aztec.js';
-import { ChildContract, ParentContract } from '@aztec/noir-contracts.js';
+import { StaticChildContract, StaticParentContract } from '@aztec/noir-contracts.js';
 
 import { setup } from './fixtures/utils.js';
 
 describe('e2e_static_calls', () => {
   let wallet: Wallet;
-  let parentContract: ParentContract;
-  let childContract: ChildContract;
+  let parentContract: StaticParentContract;
+  let childContract: StaticChildContract;
   let teardown: () => Promise<void>;
 
   beforeAll(async () => {
     ({ teardown, wallet } = await setup());
-    parentContract = await ParentContract.deploy(wallet).send().deployed();
-    childContract = await ChildContract.deploy(wallet).send().deployed();
+    parentContract = await StaticParentContract.deploy(wallet).send().deployed();
+    childContract = await StaticChildContract.deploy(wallet).send().deployed();
 
     // We create a note in the set, such that later reads doesn't fail due to get_notes returning 0 notes
     await childContract.methods.private_set_value(42n, wallet.getCompleteAddress().address).send().wait();
   });
 
   afterAll(() => teardown());
 
+  describe('direct view calls to child', () => {
+    it('performs legal private static calls', async () => {
+      await childContract.methods.private_get_value(42n, wallet.getCompleteAddress().address).send().wait();
+    });
+
+    it('fails when performing non-static calls to poorly written static private functions', async () => {
+      await expect(
+        childContract.methods.private_illegal_set_value(42n, wallet.getCompleteAddress().address).send().wait(),
+      ).rejects.toThrow('Static call cannot update the state, emit L2->L1 messages or generate logs');
```

Comment:
Please make sure that all the new error messages work both with current public and with the AVM. As we discussed, you may need to use some regexes like in

* https://github.com/AztecProtocol/aztec-packages/pull/6447
* https://github.com/AztecProtocol/aztec-packages/pull/6422

Thanks!

Reply:
The AVM test suite is separate and test per test identical to the public ones ( and as such has its own error messages) Since these are specific to the static call functionality, I would expect so simply remove the current "public" ones when we transition, so I don't think regexps are necessary.

Reply:
Updated error messages as regexp as discussed via slack!